gossip-base/src/main/java/org/apache/gossip/manager/GossipManager.java

```diff
@@ -46,14 +48,21 @@
 public abstract class GossipManager {
 
   public static final Logger LOGGER = Logger.getLogger(GossipManager.class);
+  
+  // this mapper is used for ring and user-data persistence only. NOT messages.
+  public static final ObjectMapper metdataObjectMapper = new ObjectMapper() {{
```

Comment:
I see where you are coming from with this. I do think they are all coupled. For example, if I were to send a custom type Wombat.class, I would likely need the object mapper that transmits it to be able to SerDe it and the data persister needs to SerDe it. I agree GossipManager is the catch all, but GossipCore is in my mind a fail attempt to segregate cross cutting things, so I am ok with Gossipmanager being a catch all for anything that does not clearly fit somewhere specifically.